### PR TITLE
fix test-minpoly

### PR DESCRIPTION
**Pull request by pascalgiorgi**

### CODE REVIEW

File: linbox/algorithms/massey-domain.h [hunk 1]
```diff
@@ -398,7 +398,15 @@ namespace LinBox
 		{
 			long dp = massey (phi, full_poly);
 			rank = (size_t) (v_degree(phi) - v_val (phi));
-			if (phi.size () > 0) {
+
+            if (dp==0){// zero sequence, matrix minpoly is X
```
Comment (jgdumas) [inline linbox/algorithms/massey-domain.h]: No the annihilating polynomial of a sequence is a constant

File: linbox/algorithms/massey-domain.h [hunk 2]
```diff
@@ -398,7 +398,15 @@ namespace LinBox
 		{
 			long dp = massey (phi, full_poly);
 			rank = (size_t) (v_degree(phi) - v_val (phi));
-			if (phi.size () > 0) {
+
+            if (dp==0){// zero sequence, matrix minpoly is X		
+                 std::cout<<"P="<<phi<<std::endl;
```
Comment (jgdumas) [inline linbox/algorithms/massey-domain.h]: No cout here

File: linbox/algorithms/wiedemann.h [hunk 3]
```diff
@@ -132,12 +132,12 @@ namespace LinBox
 
 //             std::cerr << "P: " << P << std::endl;
 //             std::cerr << "WD deg: " << deg << std::endl;
-        if (!deg) {
-                // zero sequence, matrix minpoly is X
-            P.resize(2);
-            A.field().assign(P[0],A.field().zero);
-            A.field().assign(P[1],A.field().one);
-        }
+        // if (!deg) {
```
Comment (jgdumas) [inline linbox/algorithms/wiedemann.h]: No managing the zero matrix should be done here

**Comment by ClementPernet** [comment]
@pascalgiorgi : any update on this one?

**Comment by pascalgiorgi** [comment]
Change has been done. The PR can be handle now.

**Pull request merged**